### PR TITLE
RFC: Ranged for

### PR DESCRIPTION
**Pull request by tautschnig**
There are still another ~1000 for loops that may be tweaked to use ranged-for. This pull request addresses a first subset of them, and also cleans up identifiers used in there.

### CODE REVIEW

**Comment by kroening** [comment]
I cherry-picked da46ebc, which seemed unrelated.

**Comment by tautschnig** [comment]
I'm updating changes and will include a regression test for the duplicate-members loop.

**Comment by tautschnig** [comment]
Rebased and comments addressed.

File: src/aa-path-symex/path_symex_state.cpp [hunk 1]
```diff
@@ -86,11 +86,8 @@ void path_symex_statet::output(const threadt &thread, std::ostream &out) const
 {
   out << "  PC: " << thread.pc << std::endl;
   out << "  Call stack:";
-  for(call_stackt::const_iterator
-      it=thread.call_stack.begin();
-      it!=thread.call_stack.end();
-      it++)
-    out << " " << it->return_location << std::endl;
+  for(const auto & call : thread.call_stack)
```
Comment (peterschrammel) [inline src/aa-path-symex/path_symex_state.cpp]: It's not very consistent to use `type & var` whereas everywhere else we use `type &var`

File: src/cpp/cpp_typecheck_compound_type.cpp [hunk 2]
```diff
@@ -647,14 +647,14 @@ void cpp_typecheckt::typecheck_compound_declarator(
...
+        code_typet::parameterst &args=code_type.parameters();
+        unsigned i=0;
+        for(auto & arg : args)
         {
-          code_typet::parametert& arg = args[i];
           irep_idt base_name = arg.get_base_name();
 
           if(base_name==irep_idt())
-            base_name="arg"+i2string(i);
+            base_name="arg"+i2string(i++);
```
Comment (peterschrammel) [inline src/cpp/cpp_typecheck_compound_type.cpp]: Ok. Good.

File: src/goto-instrument/wmm/goto2graph.cpp [hunk 3]
```diff
@@ -429,21 +425,19 @@ void inline instrumentert::cfg_visitort::visit_cfg_reference_function (
...
         {
           instrumenter.message.debug() << "t" << messaget::eom;
-          if(in_pos.find(*t)!=in_pos.end()
-            && updated.find(*t)!=updated.end())
+          if(in_pos.find(in)!=in_pos.end()
+            && updated.find(in)!=updated.end())
           {
-            /* out_pos[*t].insert(in_pos[*t])*/
-            add_all_pos(it1, out_pos[*t], in_pos[*t]);
+            /* out_pos[in].insert(in_pos[in])*/
```
Comment (peterschrammel) [inline src/goto-instrument/wmm/goto2graph.cpp]: Side note: why is there commented code?

File: src/solvers/smt1/smt1_conv.cpp [hunk 4]
```diff
@@ -3625,7 +3625,7 @@ exprt smt1_convt::binary2struct(
     index-=sub_size;
     std::string cval=binary.substr(index, sub_size);
 
-    e.operands()[i]=ce_value(sub_type, "", cval, true);
+    e.operands()[i++]=ce_value(sub_type, "", cval, true);
```
Comment (peterschrammel) [inline src/solvers/smt1/smt1_conv.cpp]: ok

**Comment by kroening** [comment]
This is starting to look good to me -- may I ask for a rebase?

File: src/aa-path-symex/path_symex_state.cpp (hunk 1 above)
Comment (tautschnig) [inline src/aa-path-symex/path_symex_state.cpp]: I'm cleaning this up and will include that in the rebased version.

**Comment by tautschnig** [comment]
@kroening rebase and cleanup done.
